src/web-server.js

```diff
@@ -118,54 +120,111 @@ app.patch("/import-map.json", (req, res) => {
     return;
   }
+  if (req.body.imports) {
+    if (typeof req.body.imports !== "object") {
+      return res
+        .status(400)
+        .send(`Invalid import map in request body -- imports is not an object`);
+    }
 
-  if (req.body.scopes) {
-    res
-      .status(400)
-      .send("import-map-deployer does not support import map scopes");
-    return;
-  }
+    if (Object.keys(req.body.imports).length === 0) {
+      res
+        .status(400)
+        .send(
+          "Invalid import map in request body -- 'imports' object required with modules in it."
+        );
+      return;
+    }
 
-  if (!req.body.imports || Object.keys(req.body.imports).length === 0) {
-    res
-      .status(400)
-      .send(
-        "Invalid import map in request body -- 'imports' object required with modules in it."
-      );
-    return;
+    for (let moduleName in req.body.imports) {
+      if (typeof req.body.imports[moduleName] !== "string") {
+        res
+          .status(400)
+          .send(
+            `Invalid import map in request body -- module with name '${moduleName}' does not have a string url`
+          );
+        return;
+      }
+    }
   }
+  if (req.body.scopes) {
+    if (getConfig().manifestFormat !== "importmap") {
+      return res
+        .status(400)
+        .send(
+          `Invalid import map in request body -- scopes are only supported with manifest format "importmap"`
+        );
+    }
 
-  for (let moduleName in req.body.imports) {
-    if (typeof req.body.imports[moduleName] !== "string") {
-      res
+    if (typeof req.body.scopes !== "object") {
+      return res
+        .status(400)
+        .send(`Invalid import map in request body -- scopes is not an object`);
+    }
+
+    if (Object.keys(req.body.scopes).length === 0) {
+      return res
         .status(400)
         .send(
-          `Invalid import map in request body -- module with name '${moduleName}' does not have a string url`
+          `Invalid import map in request body -- scopes is an object with no properties`
         );
-      return;
+    }
+
+    for (let scopeName in req.body.scopes) {
+      if (typeof req.body.scopes[scopeName] !== "object") {
+        return res
+          .status(400)
+          .send(
+            `Invalid import map in request body -- scope with name '${scopeName}' is not an object`
+          );
+      }
+
+      if (Object.keys(req.body.scopes[scopeName]).length === 0) {
+        return res
+          .status(400)
+          .send(
+            `Invalid import map in request body -- scope with name '${scopeName}' is an object with no properties`
+          );
+      }
     }
   }
 
-  const importUrls = Object.values(req.body.imports);
+  let validImportUrlPromises = Promise.resolve();
+  if (req.body.imports) {
+    // Confirm the imports are working
+    const importUrls = Object.values(req.body.imports);
+    const unsafeUrls = importUrls.map(checkUrlUnsafe).filter(Boolean);
```

Comment:
ye good one, would be good to add a test for it as well, though maybe add that in another pr.. For testing this i need to mock fetch first, because the urls resolve checker will be the first that hits, i would likely need something like: https://www.npmjs.com/package/jest-fetch-mock. (also reversing the check would be desirable i think.. unsafe urls shouldn't be needed to resolve checked)